pkg/codegen/nodejs/gen.go

```diff
@@ -1153,10 +1156,8 @@ func (mod *modContext) genIndex(exports []string) string {
 		child := mod.mod
 		if mod.compatibility == kubernetes20 {
 			// Trim version suffix from child modules. Nested versions will have their own index.ts file.
-			if match, _ := regexp.MatchString(`/v\d+((alpha|beta)\d+)?`, child); match {
-				if i := strings.LastIndex(child, "/"); i > 0 {
-					child = child[:i]
-				}
+			if match := k8sVersionRegex.FindStringIndex(child); len(match) != 0 {
+				child = child[:match[0]-1] // -1 required to remove the leading '/'
```

Comment:
The `/` was in the regex before, which should obviate the need for the `-1`--was it necessary to remove the `/` for correctness?

Reply:
I did that so the name lined up better with what it was matching, but I could think up a new name if that makes more sense.

Reply:
It is safer to have the `/` in the regex, as the code that consumes the result is assuming its presence. I'd change the name of the regex.